api/admin/controller/__init__.py

```diff
@@ -893,30 +895,51 @@ def _create_or_update_list(
             list.library = library
 
         # Test JSON viability of auto update data
-        if auto_update_query:
-            try:
-                json.loads(auto_update_query)
-            except json.JSONDecodeError:
-                self._db.rollback()
-                return INVALID_INPUT.detailed(
-                    "auto_update_query is not JSON serializable"
-                )
-        if auto_update_facets:
-            try:
-                json.loads(auto_update_facets)
-            except json.JSONDecodeError:
-                self._db.rollback()
-                return INVALID_INPUT.detailed(
-                    "auto_update_facets is not JSON serializable"
+        try:
+            auto_update_query_dict = None
+            if auto_update_query:
+                try:
+                    auto_update_query_dict = json.loads(auto_update_query)
+                except json.JSONDecodeError:
+                    raise Exception(
+                        INVALID_INPUT.detailed(
+                            "auto_update_query is not JSON serializable"
+                        )
+                    )
+
+                if entries and len(entries) > 0:
+                    raise Exception(AUTO_UPDATE_CUSTOM_LIST_CANNOT_HAVE_ENTRIES)
+                if deletedEntries and len(deletedEntries) > 0:
```

Comment:
Since `deletedEntries` is a variable  I believe it should be  renamed to`deleted_entries` to conform to PEP 8